WordPress/gutenberg
# Fix : Empty last List Element should not be displayed

## What?

Closes https://github.com/wordpress/gutenberg/issues/65876

As mentioned in the above issue. When we create a list and if we leave the last item empty. It renders a empty list item on the front-end.

## Why?

When the user adds a list item then we are saving it as list item block. But if the list-item has no content then we are saving empty list-item. Resulting in a empty list-item element

## How?

- Added a check to only save the list-item only if it has content

## Testing Instructions
- Open the Gutenberg Editor.
- Add the list block
- Add two or three items and for the last item keep it empty
- On the front-end it should not display the last list-item ( empty one)

### Testing Instructions for Keyboard


## Screenshots or screencast 
### before 
<img width="1196" alt="Screenshot 2024-10-22 at 11 21 09 PM" src="https://github.com/user-attachments/assets/6470c75e-19c0-4e98-9984-d4436abfa143">
<img width="1452" alt="Screenshot 2024-10-22 at 11 20 59 PM" src="https://github.com/user-attachments/assets/b39dacc8-cb3a-417a-bc1d-02f594c03407">

### After

<img width="1194" alt="Screenshot 2024-10-22 at 11 17 20 PM" src="https://github.com/user-attachments/assets/4d5b5a3b-1b4f-434d-868f-bd4016159bb9">

<img width="1450" alt="Screenshot 2024-10-22 at 11 17 39 PM" src="https://github.com/user-attachments/assets/29badc83-1b23-44c1-8377-76a5bfa81799">
